src/main/java/com/microsoft/aad/msal4j/TokenResponse.java

```diff
@@ -52,28 +52,28 @@ static TokenResponse parseJsonObject(final JSONObject jsonObject)
         // use an empty string in order to avoid an IllegalArgumentException from OIDCTokens.
         String idTokenValue = "";
         if (jsonObject.containsKey("id_token")) {
-            idTokenValue =  jsonObject.getAsString("id_token");
+            idTokenValue = JSONObjectUtils.getString(jsonObject, "id_token");
         }
 
         // Parse value
         String scopeValue = null;
         if (jsonObject.containsKey("scope")) {
-            scopeValue = jsonObject.getAsString("scope");
+            scopeValue = JSONObjectUtils.getString(jsonObject, "scope");
         }
 
         String clientInfo = null;
         if (jsonObject.containsKey("client_info")) {
-            clientInfo = jsonObject.getAsString("client_info");
+            clientInfo = JSONObjectUtils.getString(jsonObject, "client_info");
         }
 
         long expiresIn = 0;
         if (jsonObject.containsKey("expires_in")) {
-            expiresIn = jsonObject.getAsNumber("expires_in").longValue();
+            expiresIn = Long.parseLong(jsonObject.getAsString("expires_in"));
```

Comment:
>Long.parseLong( [](start = 24, length = 15)

should this have been a mthod in the JSONObjectUtils instead? to align behavior and parsing logic in one place?

Reply:
yes, but that method expects json value as JSON number, so fails if it is string. AAD returns those values as JSON number, B2C as JSON String,  so can not rely on JSONObjectUtils